pkg/bgpv1/manager/reconcilerv2/paths.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"context"
+	"maps"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/instance"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+)
+
+// PathMap is a map of paths indexed by the NLRI string
+type PathMap map[string]*types.Path
```

Comment:
What is the purpose of making the map values a pointer? For example, I don't see nil checks when iterating through the maps.

Reply:
I kept the same pointer semantics as in [reconclier v1](https://github.com/cilium/cilium/blob/main/pkg/bgpv1/manager/reconciler/pod_ip_pool.go#L42). 

I'll add nil checks to make sure callers are not sending out invalid paths. Eventually, if path is not correctly formated, gobgp will return an error.

Reply:
Added nil checks [here](https://github.com/cilium/cilium/compare/d0dba953fb26c11b996c710d8a29273d9448a31c..cdbadcb1b0cc4bf8ef0358aaa052285f40539aec)